### PR TITLE
Disable more flaky tests

### PR DESCRIPTION
**Pull request by bergundy**
Follow up to #6634, the following tests are still flaky:

```
TestVersioningFunctionalSuite/TestDispatchRetry
TestUpdateWorkflowSdkSuite/TestUpdateWorkflow_TimeoutWorkflowAfterUpdateAccepted
```

Also increased XDC timeout to 40 minutes since those tests occasionally time out.

### CODE REVIEW

File: .github/workflows/run-tests.yml [hunk 1]
```diff
@@ -119,7 +119,7 @@ jobs:
       - uses: actions/setup-go@v5
         if: ${{ !inputs.run_single_functional_test && !inputs.run_single_unit_test }}
         with:
-          go-version-file: 'go.mod'
+          go-version-file: "go.mod"
```
Comment (bergundy) [inline .github/workflows/run-tests.yml:122]: VS code's fault. I can revert if this bugs anyone.

File: service/matching/backlog_manager_test.go [hunk 2]
```diff
@@ -139,6 +139,8 @@ func TestReadLevelForAllExpiredTasksInBatch(t *testing.T) {
 }
 
 func TestTaskWriterShutdown(t *testing.T) {
+	t.Skip("flaky test")
```
Comment (dnr) [inline service/matching/backlog_manager_test.go:142]: this one is fixed here: https://github.com/temporalio/temporal/pull/6627
please leave this out

File: Makefile [hunk 3]
```diff
@@ -64,9 +64,9 @@ define NEWLINE
 
 endef
 
-# 30 minutes is the upper bound defined for all tests, the longer running ones at the time of writing are XDC tests. The
-# 30 minute timeout is also defined in run-tests.yml
-TEST_TIMEOUT ?= 30m
+# 40 minutes is the upper bound defined for all tests, the longer running ones at the time of writing are XDC tests. The
+# 40 minute timeout is also defined in run-tests.yml
+TEST_TIMEOUT ?= 40m
```
Comment (dnr) [inline Makefile:69]: there are two spots that need to be updated now, and the values need to be > this value. see [`2ccbe18` (#6595)](https://github.com/temporalio/temporal/pull/6595/commits/2ccbe1881bf143ddda7f0a27491a823d268481a3)

Comment (dnr) [reply Makefile:69]: (btw 6595 makes xdc tests run in < 10 minutes. I'll reduce this after that goes in.)

File: service/matching/backlog_manager_test.go [hunk 4]
```diff
@@ -139,6 +139,8 @@ func TestReadLevelForAllExpiredTasksInBatch(t *testing.T) {
...
+
```
Comment (bergundy) [inline service/matching/backlog_manager_test.go:143]:
```suggestion
```

File: Makefile (hunk 3 above)
Comment (bergundy) [reply Makefile:69]: oh wow!

File: service/matching/backlog_manager_test.go (hunk 2 above)
Comment (dnr) [reply service/matching/backlog_manager_test.go:142]: this is still here?

File: service/matching/backlog_manager_test.go (hunk 4 above)
Comment (bergundy) [inline service/matching/backlog_manager_test.go:143]:
```suggestion
```

**Pull request merged**